### DISCUSSION TITLE
What happens at year end for home assistant energy integration

### DISCUSSION THREAD

**Discussion by genestealer** [HandyHat/ha-hildebrandglow-dcc#210] [Q&A] [answered]
Sorry I did look through the previous issues and couldn't find an answer.

**Is the consumption (year) a calendar year from January to December, resetting on 1 January, or a rolling 12 month period?**

1. If it's from January to December then, what sort of behaviour are expecting to see in the home assistant energy integration on 1 January? I was under the impression home assistant uses the provided state as a long-term statistic sensor, which means if it suddenly drops to 0 I'm not sure how it will handle that.

3. If it's a rolling 12 months then I do not see how the consumption can be accurately used.

Update:
Apologies, I had a quick look and I realised that my own energy monitoring is based upon a per hour kWh sensor, which it seems to manage perfectly.  However it still be good for someone to confirm what behaviour they saw on the 1st Jan?

Update 2:
If it is the case that it doesn't matter that the sensor resets at the end of the year, then why is there any advantage over using the 'yearly consumption' vs. the 'daily consumption'.

**Comment by HandyHat** [comment, marked as answer]
The yearly consumption sensor is indeed a calendar year resetting on the 1st of January. It's state class is set to `total_increasing`, and so when it resets to 0 it simply treats it as a cycle reset.

The advantage of using the yearly sensor over the daily one is that the daily sensor misses out the last 30 minutes of each day, making the cumulative consumption inaccurate. The rational behind this can be seen in #49 .

As described in #191, I am looking into changing the daily sensor to include this missing 30 mins, negating the need for a yearly sensor.
